README.md

```diff
@@ -0,0 +1,95 @@
+## What is `copr-test`
+
+The `copr-test` is the integration test for `Coprocessor` of the TiKV. The `Coprocessor` is used to execute TiDB push-down
+executor to improve the database performance. It's important to keep execution result consistent between TiDB and `Coprocessor`.
+This is the purpose of `copr-test`.
+
+The basic principle of `copr-test` is to execute test cases in TiDB standalone (with `mocktikv`) and TiDB cluster (with `Coprocessor`)
+and compare the execution result. We think the TiDB and Coprocessor is consistent if the results in both sides are same, vice versa.
+We use `github.com/pingcap/parser` to parse every file, and iterator the []ast.StmtNode to run every SQL.
+So if there is any error in the test SQL file, the error msg is hard to read. Creating the database and tables in you local database
+and use IDE(such as CLion) to write SQL file as test case is recommended.
+
+The `Coprocessor` contains two execution frameworks, `Batch` (vectorized execution engine) and `Non-Batch` in current
+stage (remove the `Non-Batch` version is in plan). So all test cases should be passed in the following execution environment.
+
+1. TiDB with mocktikv
+2. TiDB + TiKV with Non-Batch
+3. TiDB + TiKV with Batch
+
+### How do I know which expressions will be pushed down?
+
+The `Coprocessor` does not implement all builtin functions (or implemented but not tested fully) which have implemented by TiDB,
+so we can only push down the functions that Coprocessor has implemented and fully tested. Whether a function is pushed down or not
+is determined by the function `canFuncBePushed` in TiDB [tidb-expression/expr_to_pb.go](https://github.com/pingcap/tidb/blob/a090e6be2991bf85b18fcdb096f84d41f4f6bd85/expression/expr_to_pb.go#L303)
+
+We have added a failpoint `PushDownTestSwitcher` in the function `canFuncBePushed` to hijack our customized push down condition.
+And we will push down all functions in our integration tests using `export GO_FAILPOINTS="github.com/pingcap/tidb/expression/PushDownTestSwitcher=return(\"all\")"`, see the `run_tests.sh`[./push-down-test/run-tests.sh].
+
+## How to add test cases?
+
+- Test cases location
+    
+    All test cases should be placed in `push-down-test/sql` directory (sub dir of push-down-test/sql is also allowed, you can originate
+    them in different dir.), and all SQL files will be executed in lexical order by file name.
+    So if your case want to prepare some data before running tests, you can add some suffix to the file name, eg: `xxx.1.sql` and `xxx.2.sql`.
+    
+- Test case file name convention
+
+    All test cases should have the suffix `.sql` as file extension. But you can add a `xxx.md` documentation to explain the complicated cases.
```

Comment:
How about: `Files ends with .sql suffix will be treated as test cases and other files will be ignored.`